src/main/kotlin/no/nav/familie/ks/sak/kjerne/brev/BrevPeriodeContext.kt

```diff
@@ -388,6 +391,43 @@ class BrevPeriodeContext(
                         begrunnelse = begrunnelse,
                     )
 
+                val kompetanserTilBarnIBegrunnelseTidslinjer =
+                    personerGjeldendeForBegrunnelse.filter { it.type == PersonType.BARN }
+                        .associateWith { barn -> kompetanser.filter { barn.aktør in it.barnAktører }.tilTidslinje() }
+
+                val begrunnelseGjelderSøkerOgOpphørFraForrigeBehandling = (sanityBegrunnelse.begrunnelseGjelderOpphørFraForrigeBehandling()) && gjelderSøker
+
+                val relevanteKompetanser =
+                    when (begrunnelse.begrunnelseType) {
+                        BegrunnelseType.INNVILGET,
+                        BegrunnelseType.EØS_INNVILGET,
+                        BegrunnelseType.FORTSATT_INNVILGET,
+                        BegrunnelseType.REDUKSJON,
+                        BegrunnelseType.EØS_REDUKSJON,
+                        BegrunnelseType.ENDRET_UTBETALING,
+                        BegrunnelseType.ETTER_ENDRET_UTBETALING,
+                        -> {
+                            kompetanserTilBarnIBegrunnelseTidslinjer.values
+                                .flatMap { it.klippEtterVedtaksperiode().hentVerdier() }
+                                .filterNotNull()
+                                .toSet()
+                        }
+
+                        BegrunnelseType.OPPHØR,
+                        BegrunnelseType.EØS_OPPHØR,
+                        BegrunnelseType.AVSLAG,
+                        BegrunnelseType.EØS_AVSLAG,
+                        -> {
+                            if (begrunnelseGjelderSøkerOgOpphørFraForrigeBehandling) {
+                                TODO("Må se på kompetansene i samme periode forrige behandling")
```

Comment:
Hvorfor er det forskjell på om behgrunnelse gjelder søker eller ikke?

Reply:
Godt spørsmål. Har bare kopiert logikken fra ba-sak 🤔 
![image](https://github.com/navikt/familie-ks-sak/assets/17828446/41d28f27-37b7-431d-8f4c-69f4d1884f78)

Reply:
Hmm 🤔 helt sikkert en logisk grunn, men klarer ikke skjønne det. For man skulle jo tro at hvis det var opphør fra forrige behandling skulle man se på samme periode forrige behandling uavhengig av hvilken person det var...

Reply:
Burde forresten kanskje også skrive bort fra TODO før PRen merges?

Reply:
> For man skulle jo tro at hvis det var opphør fra forrige behandling skulle man se på samme periode forrige behandling uavhengig av hvilken person det var
Ja er jo enig i dette 🤔 

> Burde forresten kanskje også skrive bort fra TODO før PRen merges
Vi har ikke noe støtte for å se på forrige behandling enn så lenge, så tror det er bedre om vi tar det i en egen PR

Reply:
Jeg er litt fristet til å beholde det som det er til vi har et konkret case med opphør fra forrige behandling å teste på